src/Lexer.php

```diff
@@ -580,14 +586,20 @@ protected function scanTableRow()
         }
 
         $line = $this->getTrimmedLine();
-        if ($line === '' || !str_starts_with($line, '|') || !str_ends_with($line, '|')) {
+        if (!str_starts_with($line, '|')) {
+            // Strictly speaking, a table row only has to begin with a pipe - content to the right
+            // of the final pipe will be ignored after we split the cells.
             return null;
         }
 
+        $rawColumns = preg_split(self::CELL_PATTERN, $line);
+        assert($rawColumns !== false);
+
+        // Safely remove elements before the first and last separators
+        array_shift($rawColumns);
+        array_pop($rawColumns);
+
```

Comment:
This is taken from [cucumber/gherkin's PHP implementation](https://github.com/cucumber/gherkin/blob/679a87e21263699c15ea635159c6cda60f64af3b/php/src/StringGherkinLine.php#L75-L79).

We will probably want to borrow more of that in due course to solve the table escaping issues, but they need to be behind a feature flag to avoid a BC break.